### PR TITLE
sanity check for div/0 case in timesync state

### PR DESCRIPTION
**Pull request by y0chanan**

### CODE REVIEW

File: src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java [hunk 1]
```diff
@@ -212,6 +220,10 @@ private long computeClusterTimeOffset(List<Long> timeSyncOffsets) {
             }
         }
 
+        if(count == 0){
+            return -1;
```
Comment (DaRoli) [inline src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java]: you could add "shouldn't ever come here"

File: src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java [hunk 2]
```diff
@@ -212,6 +220,10 @@ private long computeClusterTimeOffset(List<Long> timeSyncOffsets) {
             }
         }
 
+        if(count == 0){ // shouldn't come here under normal circumstances
```
Comment (stefaneberl) [inline src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java]: space between `if` and `(`
space between `)` and `{`

File: src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java [hunk 3]
```diff
@@ -171,13 +171,21 @@ private void handleTimeSyncResponses(BeaconSendingContext context, List<Long> ti
...
         // -> handle this case
+
         if (timeSyncOffsets.size() < TIME_SYNC_REQUESTS) {
             handleErroneousTimeSyncRequest(context);
             return;
         }
 
+        //sanity check to catch case with div/0
+        long calculatedOffset = computeClusterTimeOffset(timeSyncOffsets);
+        if(calculatedOffset < 0)
```
Comment (stefaneberl) [inline src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java]: space between `if` and `(`

File: src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java [hunk 4]
```diff
@@ -171,13 +171,21 @@ private void handleTimeSyncResponses(BeaconSendingContext context, List<Long> ti
...
+        {
```
Comment (stefaneberl) [inline src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java]: in Java `{` is moved up, unlike in C# and C/C++

**Pull request merged**